tavern_pathfinder/src/lib.rs

```diff
@@ -1,7 +1,622 @@
+    	Linguistics, 
+    	Perception, 
+    	Perform, 
+    	Profession, 
+    	Ride, 
+    	SenseMotive, 
+    	SleightOfHand, 
+    	Spellcraft,
+        Stealth,
+        Survival,
+        Swim,
+        UseMagicDevice,
+    }
+
+    #[derive(Serialize,Deserialize)]
+    pub enum SaveThrow {
+        Fortitude,
+        Reflex,
+        Will,
+    }
+
+    #[derive(Serialize,Deserialize)]
+    pub enum CharacterStat {
+    	Name,
+    	Race,
+    	Size,
+    	Height,
+    	Weight,
+    	Age,
+    	Gender,
+    	Alignment,
+    	Deity,
+    	Languages,
+        Appearance,
+    }
+
+    #[derive(Serialize,Deserialize)]
+    pub enum CombatStat {
+    	MeleeAttackBonus,
+    	RangedAttackBonus,
+    	CMB,
+    	CMD,
+    	ArmorClass,
+    	TouchAC,
+    	FlatFootedAC,
+    	InitiativeBonus,
+    	DamageReduction,
+    	SpellResistance,
+        Speed,
+        FortitudeSave,
+        ReflexSave,
+        WillSave,
+    }
+
+    #[derive(Serialize,Deserialize)]
+    pub enum CasterType {
+        Spontaneous,
+        Prepared,
+    }
+
+    #[derive(Serialize,Deserialize)]
+    pub enum ComponentType {
+        Somatic,
+        Material,
+        Verbal,
+    }
+
+    #[derive(Serialize,Deserialize)]
+    pub enum MagicSchool {    
+    	Abjuration, 
+    	Conjuration, 
+    	Divination, 
+    	Enchantment, 
+    	Evocation, 
+    	Illusion, 
+    	Necromancy, 
+	    Transmutation,
+    }
+
+    #[derive(Serialize,Deserialize)]
+    pub enum SpellRange {
+        Personal,
+        Touch,
+        Close,
+        Medium,
+        Long,
+        Unlimited,
+    }
+
+    #[derive(Serialize,Deserialize)]
+    pub enum WeaponClass {
+    	Axes, 
+    	HeavyBlades, 
+    	LightBlades, 
+    	Bows, 
+    	Close, 
+    	Crossbows, 
+    	Double, 
+    	Firearms, 
+    	Flails, 
+    	Hammers, 
+    	Monk, 
+    	Natural, 
+    	Polearms, 
+	    SiegeEngines, 
+	    Spears,
+        Thrown,
+        Tribal,
+    }
+
+    #[derive(Serialize,Deserialize)]
+    pub enum ArmorClass {
+        Light,
+        Medium,
+        Heavy,
+    }
+
+    #[derive(Serialize,Deserialize)]
+    pub enum EquipmentSlot {
+    	NoSlot,
+    	Armor, 
+    	Belts, 
+    	Body, 
+    	Chest, 
+    	Eyes, 
+    	Feet, 
+    	Hands, 
+    	Head, 
+    	Headband, 
+    	Neck, 
+    	RingLeft, 
+    	RingRight, 
+    	Shield, 
+    	Shoulders,
+        Wrist,
+    }
+
+    #[derive(Serialize,Deserialize)]
+    pub enum DamageType {
+    	Bludgeoning, 
+    	Slashing, 
+    	Piercing, 
+    	Energy, 
+    	Acid, 
+       	Fire, 
+    	Electricity, 
+    	Cold, 
+    	Sonic, 
+    	Positive, 
+    	Negative, 
+	    Nonlethal,
+    }
+}
+
+pub mod entities {
+    pub mod character {
+        use serde::{Serialize,Deserialize};
+        use crate::types;
+        use uuid::Uuid;
+
+        #[derive(Serialize, Deserialize)]
+        pub struct Character {
+            id: Uuid,
+            char_id: Uuid,
+            race_id: Uuid,
+            deity_id: Uuid,
```

Comment:
These exist as foreign keys in the database, so they really should be either copies of or references to the appropriate data type. This applies across all the other types, too.

I wasn't sure where to make this note, so I'll make it here: we're going to want REST-specific data on these as well. Including:

- `Summary` versions of the larger types, too. E.g. a `CharacterSummary` would only contain the `id`, `name`, `race`, and `level`: data you would want to see in a list of characters.
    - This will help prevent creating massive objects in memory when not needed, and will prevent unnecessary data from being included in responses.
    - We would eventually implement `TryFrom<CharacterSummary> for Character` and `(Try)From<Character> for CharacterSummary` or similar for easier conversion between the two.
    - Side note: the above impls would want to be feature-gated for server/client.
- Everything that will have a REST endpoint is also going to need a `links: HashMap<&str, Link>` member, for
  ```
  struct Link {
      rel: String,
      url: String/Url,
  }
  ```